### ISSUE TITLE
GPTScore: A Novel Framework for Evaluating Text Generation Quality

### ISSUE THREAD

**Issue by ShellLM** [irthomasthomas/undecidability#818]
- [ ] [GPTScore: A Novel Framework for Evaluating Text Generation Quality](https://paperswithcode.com/paper/gptscore-evaluate-as-you-desire)

# GPTScore: A Novel Framework for Evaluating Text Generation Quality

Assessing the quality of the generation is an even more arduous task than the generation itself, and this issue has not been given adequate consideration recently. This paper proposes a novel evaluation framework, GPTScore, which utilizes the emergent abilities (e.g., zero-shot instruction) of generative pre-trained models to score generated texts.

Given a text generated from a certain context, and desirable evaluation aspects (e.g., fluency), the high-level idea of the proposed framework is that the higher-quality text of a certain aspect will be more likely generated than unqualified ones based on the given context.

## Task Specification

- **Generate a summary for the following text.**

## Aspect Definition

- **REL:** The details provided by the generated text are consistent with the details in the source text.
- **INF:** ...

## Template

```
{Task_Specification}
{Aspect_Definition}
Text: {Text}
Tl;dr: {Summ}
```

## Evaluation Text

- **Sample Demonstrated**
  - **INF**
  - **REL**

## Samples

```
Text: ...
Summ: ...
INF
REL
```

## Generate a relevant summary with consistent details for the following text.

```
Text: {Text} Tl;dr: {Summ}
```

## Demo Sample GPTScore

- **INF:** 0.8
- **REL:** 0.9

## Evaluation Protocol

The framework of GPTScore includes two evaluation aspects: relevance (REL) and informative (INF). We use the evaluation of relevance (REL) of the text summarization task to exemplify our framework. "Likely" can be measured by the conditional generation probability. To capture users’ true desires, an evaluation protocol will be initially established based on:

- (a) the task specification, which typically outlines how the text is generated (e.g., generate a response for a human based on the conversation.)
- (b) aspect definition that documents the details of desirable evaluation aspects (e.g., the response should be intuitive to understand).

Subsequently, each evaluation sample will be presented with the evaluated protocol with optionally moderate exemplar samples, which could facilitate the model’s learning. Lastly, a large generative pre-trained model will be used to calculate how likely the text could be generated based on the above evaluation protocol, thus giving rise to our model’s name: GPTScore.

## Experimentally

We ran through almost all common natural language generation tasks in NLP, and the results showed the power of this new paradigm. The main observations are:

1. Evaluating texts with generative pre-training models can be more reliable when instructed by the definition of task and aspect, providing a degree of flexibility to accommodate various evaluation criteria. Furthermore, incorporating exemplified samples with in-context learning will further enhance the process.
2. Different evaluation aspects exhibit certain correlations. Combining definitions with other highly correlated aspects can improve evaluation performance.
3. The performance of GPT3-text-davinci-003, which is tuned based on human feedback, is inferior to GPT3-text-davinci-001 in the majority of the evaluation settings, necessitating deep explorations on the working mechanism of human feedback-based instruction learning (e.g., when it will fail).

### 2.1. Text Evaluation

Text evaluation aims to assess the quality of hypothesis text \(h\) in terms of certain aspect \(a\) (e.g., fluency), which is either measured manually with different protocols or quantified by diverse automated metrics.

### 2.2. Meta Evaluation

Meta evaluation aims to evaluate the reliability of automated metrics by calculating how well automated scores (\(y_{auto}\)) correlate with human judgment (\(y_{human}\)) using correlation functions such as spearman correlation and Pearson correlation.

### 2.3. Evaluation Strategy

Evaluation strategies define different aggregation methods when calculating the correlation scores. Specifically, for each source text \(s_i\), there are \(J\) system outputs \(h_{i,j}\), \(f_{auto}\) is an automatic scoring function, and \(f_{human}\) is the gold human scoring function. For a given evaluation aspect \(a\), the meta-evaluation metric \(F\) can be formulated as follows. Sample-level defines that a correlation value is calculated for each sample separately based on outputs of multiple systems, then averaged across all samples.

For more details, visit [GPTScore: Evaluate as You Desire](https://paperswithcode.com/paper/gptscore-evaluate-as-you-desire).

#### None
